runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java

```diff
@@ -158,4 +167,21 @@ public void onGameStateChanged(GameStateChanged event)
 			abyssObjects.clear();
 		}
 	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		Query darkMageQuery = new NPCQuery().idEquals(NpcID.DARK_MAGE);
+		NPC[] result = queryRunner.runQuery(darkMageQuery);
+		darkMage = result.length >= 1 ? result[0] : null;
+
+		Query inventoryQuery = new InventoryWidgetItemQuery().idEquals(
```

Comment:
`InventoryItemQuery` would be better here since it scans Inventories and not widgets, but it does not have an `idEquals` yet. I would probably add one to it, and use that.

Reply:
Done. Out of curiosity, can you give a high level explanation on the difference between an Item and a WidgetItem?

Reply:
One scans the internal Inventory objects in the client, the other the widget (interfaces)